docs/ENTITY_RELATIONSHIP_DIAGRAM.md

```diff
@@ -6,89 +6,94 @@ erDiagram
   CanonicalField {
     int id
     string label
-    string type
-    datetime dateCreated
+    string short_code
+    string field_type
+    timestamp created_at
   }
   Applicant {
     int id
-    string externalId 
-    bool hasGivenPermission
+    string external_id
+    bool opted_in
+    timestamp created_at
   }
   ExternalSource {
     int id
-    string name
+    string label
+    string website
+    timestamp created_at
   }
   Application {
     int id
-    int applicantId
-    int opportunityId
-    datetime dateCreated
+    int applicant_id
+    int opportunity_id
```

Comment:
Then again, I suppose we don't have a natural (external to the db) identity for each application schema, so that would require the caller to get that surrogate (internal to the db) identity first. Hmm.

Reply:
I still think something is not quite right. It looks like you could mix fields from multiple application schemas within a single application version. To resolve, either application needs to be for a certain application schema or application version needs to be for a certain application schema.

Reply:
I figured this would be enforced easily by business logic rather than at the database level (the restriction you describe is a business logic restriction, not necessarily a function of normalization)

Which is to say: we can add it, but adding it does not prevent the normalization risk.  In EITHER CASE the business logic will need to ensure that the proper application schema is used for all fields of a given version.

The application should not be linked to a schema though -- it is a response to an opportunity.